sql/src/main/java/io/crate/planner/operators/LogicalPlan.java

```diff
@@ -104,7 +104,7 @@
          *                         outputs: [_fetch, a]
          *                    </pre>
          */
-        LogicalPlan build(TableStats tableStats, Set<Symbol> usedBeforeNextFetch);
+        LogicalPlan build(TableStats tableStats, Set<PlanHint> hints, Set<Symbol> usedBeforeNextFetch);
```

Comment:
Currently there is only a single hint to tell `Collect` to prefer source lookup references instead of using doc-values. (Less memory intensive for insert-from-query operations). 

A concrete `preferSourceLookups` boolean would also do, but I felt that introducing some kind of plan hints better convey the semantics.

Reply:
This could later also become an optimization rule, but for that some other things have to change first I think.